draft-ietf-mls-protocol.md

```diff
@@ -1698,85 +1684,110 @@ struct {
     HPKEPublicKey public_key;
     Credential credential;
     Capabilities capabilities;
+    Extension extensions<V>;
```

Comment:
I don't love the aesthetics of this.  I would like to maintain the invariants that (a) extensions are always at the end of the message and (b) FooTBS is always clearly an extension of Foo.  You've done (b) here, but it's a little hidden in the `select`.  I think the original structs here are clearer, if a bit chattier.